pkg/plugin/plugin.go

```diff
@@ -366,6 +383,21 @@ func CmdAdd(args *skel.CmdArgs) error {
 		}
 
 		err = contNetns.Do(func(_ ns.NetNS) error {
+			if mac == "" && !sriov.IsOvsHardwareOffloadEnabled(netconf.DeviceID) {
+				for _, ipc := range newResult.IPs {
```

Comment:
What is the point of cycle here? We always either fail on the first one or use the first one. No? Would it be clearer if you added an extra condition `len(newResult.IPs) >= 1` and then accessed the first IP directly? So it is clear that we care about the first IP if there is any, not that we are iterating.

Reply:
oh! yes. good find :) I thought `newResult.IPs` was a map and added for loop with break. removed it now.